testbed/tests/test_window.py

```diff
@@ -479,6 +480,22 @@ async def test_full_screen(second_window, second_window_probe):
         assert not second_window_probe.is_full_screen
         assert second_window_probe.content_size == initial_content_size
 
+    @pytest.mark.parametrize("second_window_kwargs", [{}])
+    async def test_merge_all_windows(
+        app, main_window, second_window, second_window_probe
+    ):
+        if toga.platform.current_platform != "macOS":
+            pytest.xfail("Merging all windows is only implemented for macOS.")
```

Comment:
We generally try to avoid having platform-specific tests in the testbed, instead writing a generic test,. and putting the xfail somewhere on the proble when the platform doesn't support that action (e.g., an "window_probe.perform_window_merge()" command).

This is somewhat similar to best practice in browsers - rather than looking for a specific version of a browser ID string, you test for the feature you need.